### ISSUE TITLE
The entity no longer has a state class

### ISSUE THREAD

**Issue by mogons57** [megakid/ha_hildebrand_glow_ihd_mqtt#60]
Just got the following message relating to the rates reported by the IHD.

=============
We have generated statistics for 'Smart Meter Electricity: Import Standing Charge' (sensor.smart_meter_electricity_import_standing_charge) in the past, but it no longer has a state class, therefore, we cannot track long term statistics for it anymore.

Statistics cannot be generated until this entity has a supported state class.
If the state class was previously provided by an integration, this might be a bug. Please report an issue.
If you previously set the state class yourself, please correct it. [The different state classes and when to use which can be found in the developer documentation.](https://developers.home-assistant.io/docs/core/entity/sensor/#long-term-statistics)
If the state class has permanently been removed, you may want to delete the long term statistics of it from your database.
Do you want to permanently delete the long term statistics of sensor.smart_meter_electricity_import_standing_charge from your database?
================

The message applies to the the following entities:

sensor.smart_meter_electricity_import_unit_rate
sensor.smart_meter_electricity_import_standing_charge
sensor.smart_meter_gas_import_unit_rate
sensor.smart_meter_gas_import_standing_charge

Not sure what has changed in HA, or the electricity meter to have brought this on but it would be nice if we could keep the long term stats for these sensors.

**Comment by shuntfield** [comment]
Yes I have the same problem, I am going to see if  I can amend mine now.

**Comment by shuntfield** [comment]
@mogons57 
Go into sensor.py in the /homeassistant/custom_components/hildebrand_glow_ihd/ folder using Home Assistant File Editor, and find the sensors under their name. 
Change the `"state_class": none,` to `"state_class": SensorStateClass.TOTAL,` on each one that is in error, and then that will remove the repairs and resolve the problem. You will need to save the file and then restart home assistant. I have done this and the repairs are now gone.

**Comment by mogons57** [comment]
@shuntfield 
Fantastic that did the trick, back to normal again. Many thanks for the help.

**Comment by viking2010** [comment]
Going forward, can anyone submit a pull request for this change? Not that the previous two have been approved! 😄

**Comment by fanfarestreak** [comment]
Is there a way to update the code for this? I am happy to give it a try if someone can point me at 
1. the right file to update and 
2. a github guide of how to do it correctly
I've never done a github submission before but now seems a good as time as any to have a try if it's as simple as the code change specified above,.

**Comment by shuntfield** [comment]
@fanfarestreak the file to edit is in where I posted earlier in the thread.
As for PRs etc I don't know how to do that 😳

**Comment by fanfarestreak** [comment]
@shuntfield The first time I looked in the code somehow I completely missed that. Thanks for making me look again. 
If nobody else picks this up I will attempt to correct the code next week, but as I'll be learning github as I go it may not be fast

**Comment by mogons57** [comment]
This problem occurred again when I did the latest update but the same fix as before worked. Unfortunately I don't know how update the code on github to fix this permanently.

**Comment by shuntfield** [comment]
I think I have added a pull request here:
https://github.com/megakid/ha_hildebrand_glow_ihd_mqtt/pull/61

**Comment by mogons57** [comment]
> I think I have added a pull request here: #61

@shuntfield I've been looking https://developers.home-assistant.io/docs/core/entity/sensor/ and it looks to me as if these should be set as SensorStateClass.MEASUREMENT rather than TOTAL. I did change mine to MEASUREMENT and it seems to work but I don't know if that mucks up the historical values.

**Comment by shuntfield** [comment]
> > I think I have added a pull request here: #61
> 
> @shuntfield I've been looking https://developers.home-assistant.io/docs/core/entity/sensor/ and it looks to me as if these should be set as SensorStateClass.MEASUREMENT rather than TOTAL. I did change mine to MEASUREMENT and it seems to work but I don't know if that mucks up the historical values.

I can change it, see what happens! I'm not exactly sure what I am doing anyway!

**Comment by mogons57** [comment]
> > > I think I have added a pull request here: #61
> > 
> > 
> > @shuntfield I've been looking https://developers.home-assistant.io/docs/core/entity/sensor/ and it looks to me as if these should be set as SensorStateClass.MEASUREMENT rather than TOTAL. I did change mine to MEASUREMENT and it seems to work but I don't know if that mucks up the historical values.
> 
> I can change it, see what happens! I'm not exactly sure what I am doing anyway!

Ahh... the blind leading the blind!

**Comment by shuntfield** [comment]
> Ahh... the blind leading the blind!

Indeed mate! 🤣 I have modified the PR see how that looks now?

**Comment by megakid** [comment]
Apologies for delay, I've merged and released v1.7.1

**Comment by jeverley** [comment]
Just a heads up - Home Assistant doesn't support "monetary" device class sensors in conjunction with a state class of "measurement" see the excerpt below from  https://developers.home-assistant.io/docs/core/entity/sensor/?_highlight=monetary#entities-not-representing-a-total-amount

```
Entities not representing a total amount
Home Assistant tracks the min, max and mean value during the statistics period. The state_class property must be set to SensorStateClass.MEASUREMENT, and the device_class must **not** be either of SensorDeviceClass.DATE, SensorDeviceClass.ENUM, SensorDeviceClass.ENERGY, SensorDeviceClass.GAS, SensorDeviceClass.MONETARY, SensorDeviceClass.TIMESTAMP, SensorDeviceClass.VOLUME or SensorDeviceClass.WATER.
```

The presence of the state class raises a warning in the logs like below:

```
Entity sensor.smart_meter_electricity_import_unit_rate (<class 'custom_components.hildebrand_glow_ihd.sensor.HildebrandGlowMqttSensor'>) is using state class 'measurement' which is impossible considering device class ('monetary') it is using; expected None or one of 'total'; Please update your configuration if your entity is manually configured, otherwise create a bug report at https://github.com/megakid/ha_hildebrand_glow_ihd_mqtt
```

I'd recommend we remove the device class in this case (it won't prevent us from using the correct unit of measure etc).

This only affects the rate sensors because as the daily total sensors _are_ supported by the monetary class as they're a balance value.

I'll test this change locally and raise a PR to address the warning.
